rs/execution_environment/tests/storage_reservation.rs

```diff
@@ -195,20 +195,29 @@ fn test_storage_reservation_triggered_in_canister_snapshot_without_enough_cycles
 
     // Take a snapshot to trigger more storage reservation. The canister does not have
     // enough cycles in its balance, so this should fail.
-    let res = env.take_canister_snapshot(TakeCanisterSnapshotArgs::new(canister_id, None));
-    match res {
-        Ok(_) => panic!("Expected an error but got Ok(_)"),
-        Err(err) => {
-            assert_eq!(err.code(), ErrorCode::InsufficientCyclesInMemoryGrow);
-            // Match on a substring of the error message. Due to a difference in instructions consumed on
-            // Mac vs Linux, we cannot match on the exact number of cycles but we only need to verify it's
-            // a non-zero amount.
-            assert!(
-                err.description()
-                    .contains("due to insufficient cycles. At least 339_559_"),
-                "Error message: {}",
-                err.description()
-            );
-        }
-    }
+    let err = env
+        .take_canister_snapshot(TakeCanisterSnapshotArgs::new(canister_id, None))
+        .expect_err("Expected an error, but got Ok(_)");
+    err.assert_contains(
+        ErrorCode::InsufficientCyclesInMemoryGrow,
+        "Canister cannot grow memory by 200068382 bytes due to insufficient cycles.",
```

Comment:
I believe that matching on the precise amount of bytes also resulted in some flakiness on CI, not really sure why but I had decided to leave it out since it's not the focus of the test.

Reply:
Oh actually it's probably because of macos vs linux. See [here](https://github.com/dfinity/ic/actions/runs/12884161165/job/35919861641?pr=3539) for the failure.